docs/docs/spaces.mdx

```diff
@@ -0,0 +1,271 @@
+
+spaces:
+  - name: main
+    domain: main/domain.yml
+    nlu: main/nlu/training_data.yml
+    nlu_test: main/nlu/test_data.yml
+    rules: main/rules.yml
+  - name: transfer_money
+    domain: transfer_money/domain.yml
+    nlu: transfer_money/nlu/training_data.yml
+    nlu_test: transfer_money/nlu/test_data.yml
+    rules: transfer_money/rules.yml
+    entry_intents:
+      - transfer_money
+  - name: investment
+    domain: investment/domain.yml
+    nlu: investment/nlu/training_data.yml
+    nlu_test: investment/nlu/test_data.yml
+    rules: investment/rules.yml
+    entry_intents:
+      - buy_stock
+  - name: pay_cc
+    domain: pay_cc/domain.yml
+    nlu: pay_cc/nlu/training_data.yml
+    nlu_test: pay_cc/nlu/test_data.yml
+    rules: pay_cc/rules.yml
+    entry_intents:
+      - pay_cc
+
+importers:
+  - name: "rasa_plus.spaces.space_data_importer.SpaceDataImporter"
+    temporary_working_directory: spaces
+
+pipeline:
+  - name: WhitespaceTokenizer
+  - name: RegexFeaturizer
+  - name: LexicalSyntacticFeaturizer
+  - name: CountVectorsFeaturizer
+  - name: CountVectorsFeaturizer
+    analyzer: char_wb
+    min_ngram: 1
+    max_ngram: 4
+  - name: rasa_plus.spaces.components.spaces_crf_entity_extractor.SpacesCRFEntityExtractor
+  - name: EntitySynonymMapper
+  - name: DIETClassifier
+    epochs: 100
+    ranking_length: 0
+    entity_recognition: false
+    BILOU_flag: false
+  - name: rasa_plus.spaces.components.filter_and_rerank.FilterAndRerank
+  - name: FallbackClassifier
+    threshold: 0.3
+    ambiguity_threshold: 0.1
+
+policies:
+  - name: RulePolicy
+
+```
+
+In the above config file, you can see the Spaces definition with the main space, which
+is a designated name for a space for shared primitives, and the three
+subspaces for transferring money, investment, and credit card payments. The main space
+is not required, but the only way to share primitives between spaces. Each path value
+can either point to a single file or a directory.
+
+Second, you can also see the importer
+`rasa_plus.spaces.space_data_importer.SpaceDataImporter` being used. This importer
+reads the above Spaces definition and assembles the bot. The results can be seen in the
+configurable temporary working directory.
+
+Third, you can see the specific NLU components that are necessary to make Spaces work.
+The most important is the `FilterAndRerank` component which is central to make Spaces
+work by post-processing intent rankings and entity extractions. If you also need entity
+extraction you need to use the `SpacesCRFEntityExtractor` to make that work with Spaces.
+
+Finally, you can see that we are only using the `RulePolicy` here.
+
+After you have created your spaces and made adjustments to the `config.yml`, you can
+train your new assistant using `rasa train -c config.yml` and start it afterward with
+`rasa shell`.
+
+### Training only a specific subspace
+
+We have added `--space` argument to the rasa train command to give you the option to
```

Comment:
How does this affect use of the `--data` and `--domain` cli options. If those options are used and there's a spaces config are the options assumed be used for the main space or are these options ignored when a spaces config is in use?

Reply:
I replied to you on Notion:
> Indeed, the rasa train command only support either --space or --domain or --data now. Not a combination of those.

I think it's fine to leave that out of the docs for the alpha -- the CLI will yield a descriptive error if a user attempts to use them in combination